File: app/src/main/java/com/kelsos/mbrc/views/MainView.java
package com.kelsos.mbrc.views;

import android.graphics.Bitmap;
import android.support.annotation.Nullable;

import com.kelsos.mbrc.annotations.PlayerState;
import com.kelsos.mbrc.annotations.Repeat;
import com.kelsos.mbrc.domain.TrackInfo;
import com.kelsos.mbrc.enums.LfmStatus;
import com.kelsos.mbrc.events.ui.ShuffleChange;

public interface MainView extends BaseView {

  void updateCover(@Nullable Bitmap cover);

  void updateShuffleState(@ShuffleChange.ShuffleState String shuffleState);

  void updateRepeat(@Repeat.Mode String mode);

  void updateVolume(int volume, boolean mute);

  void updatePlayState(@PlayerState.State String state);

  void updateTrackInfo(TrackInfo info);

  void updateConnection(int status);

  void updateScrobbleStatus(boolean active);

  void updateLfmStatus(LfmStatus status);
}

File: app/src/main/kotlin/com/kelsos/mbrc/views/MainView.kt
package com.kelsos.mbrc.views

import com.kelsos.mbrc.annotations.PlayerState.State
import com.kelsos.mbrc.annotations.Repeat.Mode
import com.kelsos.mbrc.domain.TrackInfo
import com.kelsos.mbrc.enums.LfmStatus
import com.kelsos.mbrc.events.ui.ShuffleChange.ShuffleState
import com.kelsos.mbrc.mvp.BaseView

interface MainView : BaseView {

  fun updateShuffleState(@ShuffleState shuffleState: String)

  fun updateRepeat(@Mode mode: String)

  fun updateVolume(volume: Int, mute: Boolean)

  fun updatePlayState(@State state: String)

  fun updateTrackInfo(info: TrackInfo)

  fun updateConnection(status: Int)

  fun updateScrobbleStatus(active: Boolean)

  fun updateLfmStatus(status: LfmStatus)
}
